### ISSUE TITLE
AVRO schema refs with Hmaba avro

### ISSUE THREAD

**Issue by VipulZopSmart** [hamba/avro#308]
Team, 
The use-case here is we have a union schema registered on the topic that references all the sub-schemas.
Now, it should be something like this: we make avro connection with union-schema and then after getting the union-schema, resolve all the references from that and use it while publishing the messages, can you please let us know how's this possible to achieve here?
Example:
Union schema name - `user-value`
`
[
  "com.test.account.user_created",
  "com.test.account.user_deleted"
]
`

Sub-schemas:
user_created.avro
`
{
  "fields": [
    {
      "doc": "name of event being published",
      "name": "event_name",
      "type": "string"
    },
    {
      "doc": "unique users reference id",
      "logicalType": "uuid",
      "name": "user_ref_id",
      "type": "string"
    }
  ],
  "name": "user_created",
  "namespace": "com.test.account",
  "type": "record",
  "version": "1"
}
`

`
{
  "fields": [
    {
      "doc": "name of event being published",
      "name": "event_name",
      "type": "string"
    },
    {
      "doc": "unique users reference id",
       "name": "user_name",
      "type": "string"
    }
  ],
  "name": "user_deleted",
  "namespace": "com.test.account",
  "type": "record",
  "version": "1"
}
`

**Comment by nrwiersma** [comment]
What have you tried that is not working?

**Comment by VipulZopSmart** [comment]
How to publish the data with using union schema? Any examples around that?

**Comment by VipulZopSmart** [comment]
@nrwiersma any help on this?

**Comment by nrwiersma** [comment]
Sure, I suggest you read the README on type conversions, specifically on unions: https://github.com/hamba/avro#types-conversions
and check out the examples on using the library here: https://pkg.go.dev/github.com/hamba/avro/v2#example-package-Usage.

**Comment by VipulZopSmart** [comment]
@nrwiersma I have the code implemented, the thing is if I have 7 sub-schemas referenced under a union schema, then i'm making 7 avro conns to each of the sub-schemas and using hamba-avro package for parsing and binding the messages. But the thing is I should be able to do this with a single avro conn with union-schema(that contains the reference to all thee sub-schemas), any specific examples around that?

**Comment by nrwiersma** [comment]
No, that is a really specific thing to do, and not the "basic" case, which is what is covered in the docs. I would suspect it is no different from encoding anything other schema data.

**Comment by VipulZopSmart** [comment]
@nrwiersma How will we get to know which schema to use from the union-schema for marshalling the payload?

**Comment by nrwiersma** [comment]
It will be based on the data/payload you are encoding. If you use a map, the first and only key is used to select the type/name of the schema, if it is an interface, it will resolve the schema from the registrar. This is all explained in the README.
